solver/llbsolver/history.go

```diff
@@ -298,13 +300,28 @@ func (h *HistoryQueue) gc() error {
 			if br.Pinned {
 				return nil
 			}
-			records = append(records, &br)
+			recs, err := h.hLeaseManager.ListResources(ctx, leases.Lease{ID: h.leaseID(string(key))})
```

Comment:
Can we do this only for the first time, to validate if something went wrong in between restarts.

We should also print a warning to daemon logs if we find such a case.

Reply:
Done